build_tools/cmake/rebuild.sh

```diff
@@ -35,10 +35,5 @@ else
   mkdir build
 fi
 cd build
-"$CMAKE_BIN" -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-                      -DIREE_BUILD_COMPILER=ON \
-                      -DIREE_BUILD_TESTS=ON \
-                      -DIREE_BUILD_SAMPLES=OFF \
-                      -DIREE_BUILD_DOCS=ON \
-                      -DIREE_BUILD_PYTHON_BINDINGS=ON ..
+"$CMAKE_BIN" -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
```

Comment:
I like the idea that we should have a "standard" build that covers the default configuration options. That makes sure that if someone turns something on by default it's covered by CI. I think I left more options implicit when I first set this up, but during review someone (you?) suggested explicitly specifying the defaults. But maybe that was because at that point we had some things still disabled, so it made more sense to be explicit about what we were enabling.

> * python bindings are tested by builds that use the frontends

These builds are way slower and people frequently don't (and shouldn't) wait for them. So not sure if it's actually a good idea to remove them from here.

> * docs build is tested implicitly by doc publication, does not need to be built here

I disagree with this. Doc publication is only done occasionally. This is the only build that would check it on presubmit. Much better to get the quick feedback if this doesn't take long to build. In fact, the build has been failing for > a week (https://github.com/google/iree/issues/5124).

So given that standing up a new build is somewhat of a pain, I would vote for leaving this one as explicitly enabling things and just flip samples to `ON`

Reply:
> But maybe that was because at that point we had some things still disabled, so it made more sense to be explicit about what we were enabling.

Yep. Now that we're pretty stable I'd like to be more careful with setting options.

> These builds are way slower and people frequently don't (and shouldn't) wait for them. So not sure if it's actually a good idea to remove them from here.

There are less than 10 tests in bindings/python/, and I have a strong suspicion that few changes outside of that directory would break _only_ those builds/tests. If a change actually affects those tests differently, that change should wait for the appropriate presubmits.

Reply:
Pressed enter early...

I'm on the fence for `IREE_BUILD_DOCS`. I think `build_tools/cmake/rebuild.sh` should really be simple and not set any options. If `build_tools/kokoro/gcp_ubuntu/cmake/linux/x86-swiftshader/build.sh` wants to do something special, it shouldn't use that script. "Building the docs" with the CMake option on presubmit also didn't help catch that issue.

Reply:
My point is that that build isn't particularly carefully monitored it seems, so I wouldn't rely on it to catch issues (this one is an issue in parsing, so wouldn't be caught here).

> There are less than 10 tests in bindings/python/, and I have a strong suspicion that few changes outside of that directory would break only those builds/tests. If a change actually affects those tests differently, that change should wait for the appropriate presubmits.

But it seems silly (and unfair) that someone should have to wait for us to build all of tensorflow so they can get feedback on python stuff, which for the most part requires no building of anything.

> I think build_tools/cmake/rebuild.sh should really be simple and not set any options

Why? The explicit purpose of this script is "run things just as the CI would". It only exists as a separate script so that it can be run locally to decouple "I'm a CI VM just setting everything up for the first time" and "I want to test everything just as the CI does". I'll admit that our current collection of scripts is confusing and maybe this one should be under `build_tools/kokoro`, but I maintain that there's value in separating those two things.

Reply:
I'd really like to see these options removed here, but added back with some comments.